raft/src/main/java/org/apache/kafka/snapshot/SnapshotWriter.java

```diff
@@ -74,6 +74,7 @@ private SnapshotWriter(
             compressionType,
             serde
         );
+        initializeSnapshotWithHeader();
```

Comment:
Ideally, we shouldn't change the semantic of this type in this PR. Why are we moving this here from the `static` method?